lib/src/sv_sign.c

```diff
@@ -887,7 +925,7 @@ signed_video_set_private_key(signed_video_t *self, const char *private_key, size
   SV_CATCH()
   SV_DONE(status)
   // If ONVIF is available, call initialize Onvif.
-  if (self->onvif) {
+  if (self->onvif && status == SV_OK) {
```

Comment:
Isn't it better to put this inside the TRY/CATCH if you anyhow need to check `status`?